packages/protocol/contracts/layer2/based/anchor/ShastaAnchor.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "src/shared/based/LibSharedData.sol";
+import "../eip1559/LibEIP1559Classic.sol";
+import "./PacayaAnchor.sol";
+
+/// @title ShastaAnchor
+/// @notice Anchoring functions for the Shasta fork.
+/// @custom:security-contact security@taiko.xyz
+abstract contract ShastaAnchor is PacayaAnchor {
+    uint64 public immutable shastaForkHeight;
+
+    error InvalidForkHeight();
+
+    uint256[50] private __gap;
+
+    constructor(
+        address _resolver,
+        address _signalService,
+        uint64 _pacayaForkHeight,
+        uint64 _shastaForkHeight
+    )
+        PacayaAnchor(_resolver, _signalService, _pacayaForkHeight)
+    {
+        require(shastaForkHeight == 0 || shastaForkHeight > _pacayaForkHeight, InvalidForkHeight());
```

Comment:
When `shastaForkHeight` is `0`, shall we also enforce `_pacayaForkHeight` and `_ontakeForkHeight ` to be `0`?

Reply:
Please check the last commit https://github.com/taikoxyz/taiko-mono/pull/19183/commits/6b21053e7f4bb2ec2bf7e564c43f08fb387ed2cb

Reply:
If shastaForkHeight == 0, it means shasta fork is not schedule to be active, it has to be 1 if we want to use shasta right after genesis.

Reply:
If shastaForkHeight == 0, it means shasta fork is not schedule to be active,  but _pacayaForkHeight may be scheduled to be active or is active already.

Due to historical reasons, there is no _ontakeForkHeight on L2.